File: scripts/test_solutions/355_design_twitter.js
var Twitter = function() { this.tweets = []; this.follows = new Map(); this.time = 0; };
Twitter.prototype.postTweet = function(userId, tweetId) { this.tweets.push({ userId, tweetId, time: this.time++ }); };
Twitter.prototype.getNewsFeed = function(userId) {
    const following = this.follows.get(userId) || new Set();
    following.add(userId);
    return this.tweets.filter(t => following.has(t.userId)).sort((a, b) => b.time - a.time).slice(0, 10).map(t => t.tweetId);
};
Twitter.prototype.follow = function(followerId, followeeId) {
    if (!this.follows.has(followerId)) this.follows.set(followerId, new Set());
    this.follows.get(followerId).add(followeeId);
};
Twitter.prototype.unfollow = function(followerId, followeeId) {
    if (this.follows.has(followerId)) this.follows.get(followerId).delete(followeeId);
};

File: scripts/test_solutions/355_design_twitter.py
from collections import defaultdict

class Twitter:
    def __init__(self):
        self.tweets = []
        self.follows = defaultdict(set)
        self.time = 0

    def postTweet(self, userId, tweetId):
        self.tweets.append((self.time, userId, tweetId))
        self.time += 1

    def getNewsFeed(self, userId):
        following = self.follows[userId] | {userId}
        feed = [(t, uid, tid) for t, uid, tid in self.tweets if uid in following]
        feed.sort(reverse=True)
        return [tid for _, _, tid in feed[:10]]

    def follow(self, followerId, followeeId):
        self.follows[followerId].add(followeeId)

    def unfollow(self, followerId, followeeId):
        self.follows[followerId].discard(followeeId)
